### DISCUSSION TITLE
Thanks to whomever added the kind column to the symbol picker!

### DISCUSSION THREAD

**Discussion by James-Firth** [helix-editor/helix#11338] [General] [answered]
I just upgraded from `9c479e6d` to `229784cc` and noticed there's  now 2 columns "kind" and "name" in the symbol picker pop up.

I took a quick skim to try and find the changes in the changelog https://github.com/helix-editor/helix/compare/9c479e6d..229784cc But couldn't find it (and I've barely touched Rust so didn't want to dive into the code too much)

Again it's super helpful and looks great, thanks mystery contributor!

**Comment by the-mikedavis** [comment, marked as answer]
Glad you enjoy it! That new column comes from #9647. Refactoring the picker as a table has been a goal for a while (since https://github.com/helix-editor/helix/pull/2814 and https://github.com/helix-editor/helix/pull/3053 at least) and builds on some lovely changes from @sudormrfbin. Also included in that change is some refactors to the way asynchronous actions like preview syntax highlighting and dynamic actions like workspace symbol search work and interactive global search (made possible by @pascalkuthe's excellent work in https://github.com/helix-editor/helix/pull/8021).

Symbol kind has been something we've wanted to add to the picker for a while. The original issue is in https://github.com/helix-editor/helix/issues/5446 and we saw a few PRs for it based on the old Picker layouts - https://github.com/helix-editor/helix/pull/2452 and https://github.com/helix-editor/helix/pull/9774 at least (I think there are more) - but we wanted to figure out how to keep the UI clean (basically not concatenate the symbol kind into the old row text).

Also do check out the new documentation for how to use columns in pickers at https://docs.helix-editor.com/master/pickers.html added by @rcorre in https://github.com/helix-editor/helix/pull/11218 and try out the new global search that replaced `<space>/`.

**Reply by James-Firth** [reply]
Wow thanks for the detailed reply!

I spotted those docs when looking for the change, very cool!

Loving the continued progress on Helix and hoping I could contribute at some point as an excuse to learn some Rust :)
